projects/step-frontend/src/lib/modules/grid/components/agent-list/agent-list.component.html

```diff
@@ -0,0 +1,58 @@
+<section class="header">
+  <step-btn-group>
+    <step-autorefresh-toggle [interval]="5000" (refresh)="loadTable()"> </step-autorefresh-toggle>
+  </step-btn-group>
+</section>
+<step-table [dataSource]="searchableAgent$" [inProgress]="inProgress" matSort [matSortDisableClear]="true">
+  <ng-container matColumnDef="url" stepSearchCol>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header>Url</th>
+    <td mat-cell *matCellDef="let element">
+      {{ element.agentRef!.agentUrl! }}
+    </td>
+  </ng-container>
+  <ng-container matColumnDef="type" stepSearchCol>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header>Type</th>
+    <td mat-cell *matCellDef="let element">
+      <span class="agent-type" [class]="'agent-type-'.concat(element.agentRef!.agentType)"></span>
```

Comment:
Function call inside binding.
I would suggest to use additional pipe, that will form a required class string and then use it like:
```
<span [ngClass]="element.agentRef!.agentType | agentTypeClass"></span>
```